libp2p/interface.ss

```diff
@@ -36,12 +45,11 @@
                (protos :~ protocol-or-list?))
   => Stream
 
-  ;; handler: lambda (Stream)
-  (set-protocol-handler! (proto-or-list :~ protocol-or-list? )
+  (set-protocol-handler! (proto-or-list :~ protocol-or-list?)
                          (handler : :procedure))
-  => :void)
+  => :void
 
-(defrule (protocol-or-list? obj)
-  (or (string? obj)
-      (and (list? obj)
-           (andmap string? obj))))
+  ;; Removal of protocol handlers
+  (remove-protocol-handler! (pid :~ string?))
```

Comment:
this is general something that is very rarely done -- I don't think we even allow it in go-libp2p, so probably not worth the trouble.